src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,270 @@
+package org.folio.tlib.postgres.impl;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.tlib.postgres.PgCqlField;
+import org.folio.tlib.postgres.PgCqlQuery;
+import org.z3950.zing.cql.CQLBooleanNode;
+import org.z3950.zing.cql.CQLNode;
+import org.z3950.zing.cql.CQLParseException;
+import org.z3950.zing.cql.CQLParser;
+import org.z3950.zing.cql.CQLPrefixNode;
+import org.z3950.zing.cql.CQLSortNode;
+import org.z3950.zing.cql.CQLTermNode;
+
+public class PgCqlQueryImpl implements PgCqlQuery {
+  private static final Logger log = LogManager.getLogger(PgCqlQueryImpl.class);
+
+  final CQLParser parser = new CQLParser(CQLParser.V1POINT2);
+  final Map<String, PgCqlField> fields = new HashMap<>();
+
+  String language = "english";
+  CQLNode cqlNodeRoot;
+
+  @Override
+  public void parse(String query) {
+    if (query == null) {
+      cqlNodeRoot = null;
+    } else {
+      try {
+        log.debug("Parsing {}", query);
+        cqlNodeRoot = parser.parse(query);
+      } catch (CQLParseException | IOException e) {
+        throw new IllegalArgumentException(e.getMessage());
+      }
+    }
+  }
+
+  @Override
+  public String getWhereClause() {
+    return whereHandle(cqlNodeRoot);
+  }
+
+  static String basicOp(CQLTermNode termNode) {
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return "=";
```

Comment:
This seems like a missed opportunity to force an exact-equality check, as RMB does. For example, searches in inventory for `hrid==12345` are fifty times faster than for `hrid=12345`. (I know this because I keep accidentally doing the former when exercising the Z39.50 server.)

Reply:
That's because RMB uses fulltext search for identifiers when using = . This module does NOT, after we now have TEXT and FULLTEXT (see other comment).

Reply:
OK, makes sense.